src/base/Syntax.ml

```diff
@@ -698,6 +698,8 @@ module ScillaSyntax (SR : Rep) (ER : Rep) = struct
     | MatchStmt of ER.rep ident * (pattern * stmt_annot list) list
     | ReadFromBC of ER.rep ident * string
     | AcceptPayment
+    (* forall l p *)
+    | ListIter of ER.rep ident * SR.rep ident
```

Comment:
I'd prefer it the AST constructor was called `Iterate` instead. At some point in the future we will allow user-defined inductive types, and then `forall` should be applicable to those types as well.